### DISCUSSION TITLE
LeaderboardAPI.GetRangeRank 사용문제

### DISCUSSION THREAD

**Discussion by x4cf** [naverz/zepeto-studio-kor#143] [Issue Report (Legacy)]
LeaderboardAPI.GetRangeRank 함수가 정상적으로 사용 불가능한것으로 보입니다

<img width="1730" alt="스크린샷 2021-12-26 오후 1 31 50" src="https://user-images.githubusercontent.com/90679926/147398929-71e3124d-78b4-4130-9ffa-ecd7680afad6.png">

**Comment by hyeonjin-jo** [comment]
안녕하세요.

현재 GetRangeRank함수가 정상적으로 동작하는 것을 확인했습니다.

스크린샷 올려주신 것을 확인해보니 결과 콜백에서 result변수를 활용하지 않고 계신 것 같습니다.

아래 예시 코드 첨부 드립니다.

```typescript
OnResult(result: GetRangeRankResponse) {
        Debug.Log(`result.isSuccess: ${result.isSuccess}`);
        if (result.rankInfo.myRank) {
            Debug.Log(`member: ${result.rankInfo.myRank.member}, rank: ${result.rankInfo.myRank.rank}, score: ${result.rankInfo.myRank.score}`)
        }

        if (result.rankInfo.rankList) {
            for (let i = 0; i < result.rankInfo.rankList.length; ++i) {
                var rank = result.rankInfo.rankList.get_Item(i);
                Debug.Log(`i: ${i}, member: ${rank.member}, rank: ${rank.rank}, score: ${rank.score}`);
            }
        }
    }
```

**Reply by x4cf** [reply]
테스트 코드는 잘 동작합니다! 함수 외부의 변수를 가져오려고하면 선언되지 않았다고 나오는데 외부 변수를 가져오는 방법이 있을까요?

**Reply by hyeonjin-jo** [reply]
외부 변수 관련 문제를 발견해서 아래의 링크에 답변 추가했습니다.

확인 감사합니다.

https://github.com/naverz/zepeto-studio-kor/discussions/137
